client/notification.js

```diff
@@ -345,15 +345,38 @@ _.extend(NotifyClient.prototype, {
   /**
   *
   * @param {Buffer} fileData
-  * @param {Boolean} isCsv
+  * @param {object} options
+  * @param {Boolean} confirmEmailBeforeDownload
+  * @param {string} retentionPeriod
   *
   * @returns {Dictionary}
   */
-  prepareUpload: function(fileData, isCsv = false) {
-    return {
+  prepareUpload: function(fileData, options) {
+    let data = {
       'file': _check_and_encode_file(fileData, 2),
-      'is_csv': isCsv
+      'is_csv': false
     }
+
+    if (options !== undefined) {
+      if (typeof(options) === 'boolean') {
+        data['is_csv'] = options
+      }
+      else {
+        if (options['isCsv'] !== undefined) {
+          data['is_csv'] = options['isCsv'];
```

Comment:
Why would someone do `prepareUpload(fileData, {})`? Wouldn't they just exclude the `{}` if they weren't setting anything? But yes, if they did include an empty options thing it would end up undefined that's true.

Reply:
Oh yeah duh, my bad!

Reply:
Wait no this should be fine, we set `is_csv: false` by default at the top and this shouldn't overwrite anything is `isCsv` isn't provided?

Reply:
ah you're right. i missed that false default. yes, so `is_csv` will _always_ be sent across right now.

i'm happy to keep current behaviour but i'm more interested in keeping the code readable than i am preserving behaviour if we're confident it won't break.

i think i would rather, in general, we send across fields explicitly null than don't send them at all. i dont think we should exclude not sending them across, but i prefer being explicit